contracts/hub/router/src/execute.rs

```diff
@@ -249,6 +231,14 @@ pub fn execute_withdraw_voucher(
     cross_chain_addresses: Vec<CrossChainUserWithLimit>,
     timeout: Option<u64>,
 ) -> Result<Response, ContractError> {
+    verify_cross_chain_addresses(
```

Comment:
Wouldn't it be more efficient to just let the function take`CrossChainUserWithLimit` and ignore the limit in the function?